src/tools/spin/spinquic.cpp

```diff
@@ -442,6 +443,30 @@ void SpinQuicSetRandomConnectionParam(HQUIC Connection)
     Helper.Apply(Connection);
 }
 
+const uint32_t ParamCounts[] = {
+    QUIC_PARAM_GLOBAL_LOAD_BALACING_MODE + 1,
+    QUIC_PARAM_REGISTRATION_CID_PREFIX + 1,
+    QUIC_PARAM_SESSION_SERVER_RESUMPTION_LEVEL + 1,
+    QUIC_PARAM_LISTENER_STATS + 1,
+    QUIC_PARAM_CONN_DISABLE_1RTT_ENCRYPTION + 1
+};
+
+void SpinQuicGetRandomParam(HQUIC Connection)
+{
+    QUIC_PARAM_LEVEL Level = (QUIC_PARAM_LEVEL)GetRandom(5);
+    uint32_t Param = (uint32_t)GetRandom(ParamCounts[Level]);
+
+    uint8_t OutBuffer[512];
+    uint32_t OutBufferLength = (uint32_t)GetRandom(sizeof(OutBuffer) + 1);
```

Comment:
Is the goal to have `OutBufferLength` potentially be 1 more than the size of the buffer?  I think you want the `+1` outside the `GetRandom` call.

Reply:
No. That's not the point. The goal is for it to be between 0 and `sizeof(OutBuffer)` **inclusively**.